flight/mcu/lib/HistoryBuffer/HistoryBuffer.h

```diff
@@ -0,0 +1,144 @@
+            for(unsigned i = 0; i < size/2; i++){
+                //add the value of the queue at "i"th index from the front index (frontIndex + i)
+                //if our index goes over, we mod it with the length to get it to wrap back around
+                count += arr[(frontIndex + i)%size];
+            }
+            return count/(size/2);
+        }
+
+        float getPastAverage(){
+            float count = 0.0;
+            //want to average the last 50 values (start at 50, end at 99)
+            for(unsigned i = size/2; i < size; i++){
+                //add the value of the queue at "i"th index from the front index (frontIndex + i)
+                //if our index goes over, we mod it with the length to get it to wrap back around
+                count += arr[(frontIndex + i)%size];
+            }
+            return count/(size/2);
+        }
+
+        float getCurrentSecondDerivativeAverage(){
+            //we want to get velocity values from the altitude, which means that we need the change in altitidue over the change in time
+            //we can calculate the change in velocity between the current value and the next value and divide it by the time 
+            //the array will be 50-1 in length
+            float firstDiff[(size/2)-1];
+            for(unsigned i = 0; i < (size/2)-1; i++){
+                //we do arr[index] - arr[index+1] because the lower index is going to be the most recent value, which is why it goes first
+                int currentIndex = (frontIndex + i)%size;
+                int pastIndex = (frontIndex + i+1)%size;
+
+                if(timestampArr[currentIndex]-timestampArr[pastIndex] != 0){
+                    firstDiff[i] = (arr[currentIndex] - arr[pastIndex])/((timestampArr[currentIndex]-timestampArr[pastIndex])*0.001);
+                }
+                else{
+                    firstDiff[i] = (arr[currentIndex] - arr[pastIndex])/(0.02);
+                }
+                
+            }
+
+            //we want to get acceleration from velocity, so we do the same thing but this time with the velocity array
+            //the new array with be length 49-1
+            float secondDiff[(size/2)-2];
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                if(timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size] != 0){
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/((timestampArr[(frontIndex + i)%size]-timestampArr[(frontIndex + i+1)%size])*0.001);
+                }
+                else{
+                    secondDiff[i] = (firstDiff[i] - firstDiff[i+1])/(0.02);
+                }
+
+            }
+
+            //we then calculate the average of the acceleration array
+            float avg = 0.0;
+            for(unsigned i = 0; i < (size/2)-2; i++){
+                
+                avg += secondDiff[i];
+            }
+            return(avg/float(((size/2)-2))); 
```

Comment:
This returns acceleration in units of m/s^2. We either need to change the threshold altimeter acceleration values to use m/s^2, or divide this number by 9.81 to return the acceleration in gs.